File: 05_design/src/types/Product.ts
export interface Product {
  id: string;
  name: string;
  description: string | null;
  price: number;
  stock: number;
  createdAt: string;
  updatedAt: string;
}

export interface ProductFormData {
  name: string;
  description: string;
  price: string;
  stock: string;
}

export interface CreateProductRequest {
  name: string;
  description: string | null;
  price: number;
  stock: number;
}

export interface UpdateProductRequest {
  name: string;
  description: string | null;
  price: number;
  stock: number;
}

File: 03_python_fastapi_project/src/python_fastapi_project/domain/models/product.py
from sqlalchemy import Column, Integer, String, Numeric
from .base import BaseAudit

class Product(BaseAudit):
    __tablename__ = "products"

    name = Column(String(255), nullable=False)
    description = Column(String(500), nullable=True)
    stock = Column(Integer, nullable=False)
    price = Column(Numeric(10, 2), nullable=False)
